crates/trigger/src/lib.rs

```diff
@@ -135,10 +138,11 @@ impl<Executor: TriggerExecutor> TriggerExecutorBuilder<Executor> {
                     &mut builder,
                     spin_config::ConfigHostComponent::new(runtime_config.config_providers()),
                 )?;
-            }
+                Some(builder.add_host_component(wasi_cloud::WasiCloudComponent)?)
+            };
```

Comment:
I think #1561 + `TriggerExecutor::configure_engine` should allow some of this to be simplified.